### DISCUSSION TITLE
list with conditional children

### DISCUSSION THREAD

**Discussion by data-handler** [redom/redom#470] [Q&A]
I have two components: TableCell and CheckboxTableCell

The first renders plain text, the second renders a checkbox (both render markup specific to Material Design Web components).

I also have a TableRow component with a list of TableCell children: 

`this.cells = list(this.el, TableCell);`

But I want the first cell in each row to be a CheckboxTableCell, not a TableCell.

How would I go about this?

**Comment by joona** [comment]
Second argument of the `list` -function can also be a factory function. This function receives `initData, itemData, index, listData` as arguments.

Since you get both the index and data as an argument for the factory, you can return different components from the factory based on the position or item data.

https://jsfiddle.net/75u9nhk8/

https://github.com/redom/redom/issues/173#issuecomment-579222237
